### PR TITLE
Migrate test_config.py to pytest

### LINKED ISSUE #3593: Convert bodhi/tests/server/test_config.py to PyTest
[bodhi/tests/server/test_config.py](https://github.com/fedora-infra/bodhi/blob/develop/bodhi/tests/server/test_config.py) still contains classes that derive from `unittest.TestCase`. We use PyTest, so they could be migrated to not derive from `unittest.TestCase`, or derive from `bodhi.tests.server.base.BasePyTestCase` in case they need to use the test database or WSGI app. 

This also allows the use of the simpler assert constructs instead of the self.assert*() methods. Instead of unittest `setUp()` or `tearDown()` methods, PyTest uses `setup_method(...)` or `teardown_method(...)`, or analog methods to setup/teardown stuff for different scopes.

The following PR is an example of the changes that might need to be done to this file:
#3612

### PR DESCRIPTION
**Pull request by dimitraz**
Removed dependency on unittest.TestCase.
Fixes: https://github.com/fedora-infra/bodhi/issues/3593

Signed-off-by: dimitraz <dzuccare@redhat.com>

### CODE REVIEW

**Review by ryanlerch — APPROVED** [review]
LGTM! Thanks!

File: bodhi/tests/server/test_config.py [hunk 1]
```diff
@@ -16,118 +16,117 @@
...
 
+import pytest
 from unittest import mock
-import unittest
-
 from bodhi.server import config
 
 
-class BodhiConfigGetItemTests(unittest.TestCase):
+class BodhiConfigGetItemTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestBodhiConfigGetItem"

File: bodhi/tests/server/test_config.py [hunk 2]
```diff
@@ -16,118 +16,117 @@
...
+        assert self.config.load_config.call_count == 0
 
     def test_missing(self):
         """Assert you still get normal dictionary errors from the config."""
-        self.assertRaises(KeyError, self.config.__getitem__, 'somemissingkey')
+        pytest.raises(KeyError, self.config.__getitem__, 'somemissingkey')
 
 
-class BodhiConfigGetTests(unittest.TestCase):
+class BodhiConfigGetTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestBodhiConfigGet" ...

File: bodhi/tests/server/test_config.py [hunk 3]
```diff
@@ -16,118 +16,117 @@
...
+        assert self.config.load_config.call_count == 0
 
     def test_missing(self):
         """Assert you get ``None`` when the key is missing."""
-        self.assertEqual(None, self.config.get('somemissingkey'))
+        assert self.config.get('somemissingkey') is None
 
 
-class BodhiConfigPopItemTests(unittest.TestCase):
+class BodhiConfigPopItemTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestBodhiConfigPopItem"

File: bodhi/tests/server/test_config.py [hunk 4]
```diff
@@ -16,118 +16,117 @@
...
+        pytest.raises(KeyError, self.config.pop, 'password')
 
     def test_get_missing(self):
         """Assert you still get normal dictionary errors from the config."""
-        self.assertRaises(KeyError, self.config.pop, 'somemissingkey')
+        pytest.raises(KeyError, self.config.pop, 'somemissingkey')
 
 
-class BodhiConfigCopyTests(unittest.TestCase):
+class BodhiConfigCopyTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestBodhiConfigCopy"

File: bodhi/tests/server/test_config.py [hunk 5]
```diff
@@ -16,118 +16,117 @@
...
         self.config.loaded = True
 
-        self.assertEqual({'password': 'hunter2'}, self.config.copy())
-        self.assertEqual(0, self.config.load_config.call_count)
+        assert {'password': 'hunter2'} == self.config.copy()
+        assert self.config.load_config.call_count == 0
 
 
-class BodhiConfigLoadConfig(unittest.TestCase):
+class BodhiConfigLoadConfig:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestBodhiConfigLoad"

File: bodhi/tests/server/test_config.py [hunk 6]
```diff
@@ -184,7 +182,7 @@ def test_get_config_unable_to_find_file(self, exists, log_error):
         log_error.assert_called_once_with("Unable to find configuration to load!")
 
 
-class BodhiConfigLoadDefaultsTests(unittest.TestCase):
+class BodhiConfigLoadDefaultsTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestBodhiConfigLoadDefaults"

File: bodhi/tests/server/test_config.py [hunk 7]
```diff
@@ -193,10 +191,10 @@ def test_load_defaults(self):
 
         c._load_defaults()
 
-        self.assertEqual(c, {'one': 'default'})
+        assert c == {'one': 'default'}
 
 
-class BodhiConfigValidate(unittest.TestCase):
+class BodhiConfigValidate:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestBodhiConfigValidate"

File: bodhi/tests/server/test_config.py [hunk 8]
```diff
@@ -257,162 +255,160 @@ def test_valid_config(self):
         c._validate()
 
 
-class GenerateListValidatorTests(unittest.TestCase):
+class GenerateListValidatorTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestGenerateListValidator"

File: bodhi/tests/server/test_config.py [hunk 9]
```diff
@@ -257,162 +255,160 @@ def test_valid_config(self):
...
+        with pytest.raises(ValueError) as exc:
             config._generate_list_validator()({'lol': 'wut'})
 
-        self.assertEqual(str(exc.exception),
-                         '"{\'lol\': \'wut\'}" cannot be interpreted as a list.')
+        assert str(exc.exception) == '"{\'lol\': \'wut\'}" cannot be interpreted as a list.'
 
 
-class ValidateBoolTests(unittest.TestCase):
+class ValidateBoolTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestValidateBool"

File: bodhi/tests/server/test_config.py [hunk 10]
```diff
@@ -257,162 +255,160 @@ def test_valid_config(self):
...
 
     def test_string_truthy(self):
         """Test with "truthy" strings."""
         for s in ('t', 'true', 'y', 'yes', 'on', '1'):
-            self.assertTrue(config._validate_bool(s) is True)
+            assert config._validate_bool(s)
 
 
-class ValidateNoneOrTests(unittest.TestCase):
+class ValidateNoneOrTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestValidateNoneOr"

File: bodhi/tests/server/test_config.py [hunk 11]
```diff
@@ -257,162 +255,160 @@ def test_valid_config(self):
...
         result = config._validate_none_or(str)('unicode?')
 
-        self.assertEqual(result, 'unicode?')
-        self.assertTrue(isinstance(result, str))
+        assert result == 'unicode?'
+        assert isinstance(result, str)
 
 
-class ValidatePathTests(unittest.TestCase):
+class ValidatePathTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestValidatePath"

File: bodhi/tests/server/test_config.py [hunk 12]
```diff
@@ -257,162 +255,160 @@ def test_valid_config(self):
...
         result = config.validate_path(__file__)
 
-        self.assertEqual(result, __file__)
-        self.assertTrue(isinstance(result, str))
+        assert result == __file__
+        assert isinstance(result, str)
 
 
-class ValidateRstrippedStrTests(unittest.TestCase):
+class ValidateRstrippedStrTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestValidateRstrippedStr"

File: bodhi/tests/server/test_config.py [hunk 13]
```diff
@@ -257,162 +255,160 @@ def test_valid_config(self):
...
     def test_without_trailing_slash(self):
         """With no trailing slash, the string should be returned as is."""
         result = config._validate_rstripped_str('this/should/stay/the/same')
 
-        self.assertEqual(result, 'this/should/stay/the/same')
+        assert result == 'this/should/stay/the/same'
 
 
-class ValidateSecretTests(unittest.TestCase):
+class ValidateSecretTests:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestValidateSecret"

File: bodhi/tests/server/test_config.py [hunk 14]
```diff
@@ -257,162 +255,160 @@ def test_valid_config(self):
...
         result = config._validate_secret('secret')
 
-        self.assertEqual(result, 'secret')
-        self.assertTrue(isinstance(result, str))
+        assert result == 'secret'
+        assert isinstance(result, str)
 
 
-class ValidateTLSURL(unittest.TestCase):
+class ValidateTLSURL:
```
Comment (nphilipp) [inline bodhi/tests/server/test_config.py]: --> "TestValidateTLSURL"

**Review by nphilipp — DISMISSED** [review]
Sorry for not documenting this in the issue -- I only noticed it now -- but for PyTest, the class has to be named "Test<something>". Would you please rename the test classes accordingly? I've added suggestions for each one. Thanks!

**Comment by dimitraz** [comment]
Hey @nphilipp, oops sorry about that. I'll try get to that as soon as possible!

**Comment by dimitraz** [comment]
Should be done now

**Review by ryanlerch — CHANGES_REQUESTED** [review]
Looking good now, thanks for implementing the changes suggested by @nphilipp !

However, your second commit now does not have the signed off line, which is making the checks fail. Would you be able to squash your two commits into one (and make sure this has the signed-off-by line too)

Additionally, the pip-flake8 check is failing on a few style issues that need to be fixed:

```
[2019-10-28T23:06:40.852Z] ./bodhi/tests/server/test_config.py:20:1: I100 Import statements are in the wrong order. 'from unittest import mock' should be before 'import pytest' and in a different group.
[2019-10-28T23:06:40.852Z] from unittest import mock
[2019-10-28T23:06:40.852Z] ^
[2019-10-28T23:06:40.852Z] ./bodhi/tests/server/test_config.py:20:1: I201 Missing newline between import groups. 'from unittest import mock' is identified as Stdlib and 'import pytest' is identified as Third Party.
[2019-10-28T23:06:40.852Z] from unittest import mock
[2019-10-28T23:06:40.852Z] ^
[2019-10-28T23:06:40.852Z] ./bodhi/tests/server/test_config.py:21:1: I201 Missing newline between import groups. 'from bodhi.server import config' is identified as Application and 'from unittest import mock' is identified as Stdlib.
[2019-10-28T23:06:40.852Z] from bodhi.server import config
[2019-10-28T23:06:40.852Z] ^
[2019-10-28T23:06:40.852Z] ./bodhi/tests/server/test_config.py:162:32: W504 line break after binary operator
[2019-10-28T23:06:40.852Z]         assert (str(exc.value) ==
[2019-10-28T23:06:40.852Z]                                ^
[2019-10-28T23:06:40.852Z] ./bodhi/tests/server/test_config.py:163:13: E128 continuation line under-indented for visual indent
[2019-10-28T23:06:40.852Z]             ('Invalid config values were set: \n\tauthtkt.secure: "not a bool" cannot be '
[2019-10-28T23:06:40.852Z]             ^
[2019-10-28T23:06:40.852Z] ./bodhi/tests/server/test_config.py:300:48: W291 trailing whitespace
[2019-10-28T23:06:40.852Z]         assert not config._validate_bool(False) 
```

cheers!

**Comment by dimitraz** [comment]
@ryanlerch Sorry about that, I completely forgot to get round to your comment. Is there anything else you need me to do or is that done now?

**Review by ryanlerch — APPROVED** [review]

**Comment by ryanlerch** [comment]
> @ryanlerch Sorry about that, I completely forgot to get round to your comment. Is there anything else you need me to do or is that done now?

@dimitraz No problems! I fixed the handful of issues, so after the tests pass, it should be right to be merged! Thanks!

**Pull request merged**